sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2367,6 +2367,27 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         case agg: AggregateFunction =>
           // Note: PythonUDAF does not support these advanced clauses.
           if (agg.isInstanceOf[PythonUDAF]) checkUnsupportedAggregateClause(agg, u)
+          val newAgg = agg match {
+            case idf: SupportsOrderingWithinGroup if u.orderingWithinGroup.isDefined =>
```

Comment:
This is not future-proof. `AggregateWindowFunction` extends `AggregateFunction`, but we are not performing the DISTINCT check for it.

Reply:
Good catch!